client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -250,12 +250,32 @@ public void flush() {
                         // available writers.
                         success = false;
                         log.warn("Flush on segment {} failed due to {}, it will be retried.", writer.getSegmentName(), e.getMessage());
+                        tryWaitForSuccessors();
```

Comment:
The same issue can happen during the invocation of `io.pravega.client.stream.impl.EventStreamWriterImpl#close`

Reply:
Added